ouroboros-network-api/src/Ouroboros/Network/PeerSelection/RelayAccessPoint.hs

```diff
@@ -53,6 +57,40 @@ data RelayAccessPoint = RelayAccessDomain  !DNS.Domain !Socket.PortNumber
                       | RelayAccessAddress !IP.IP      !Socket.PortNumber
   deriving (Eq, Ord)
 
+-- | These instances are used to serialize 'LedgerPeerSnapshot'
+-- consensus LocalStateQuery server which uses these instances
+-- for all its query responses. It appears they provide some improved
+-- debugging diagnostics over Serialize instances.
+instance ToCBOR RelayAccessPoint where
+  toCBOR = mconcat . (encodeListLen 3 :) . \case
+    RelayAccessDomain domain port ->
+      [encodeWord8 0, serialize' port, toCBOR domain]
+    RelayAccessAddress (IP.IPv4 ip4) port ->
+      [encodeWord8 1, serialize' port, toCBOR (IP.fromIPv4 ip4)]
+    RelayAccessAddress (IP.IPv6 ip6) port ->
+      [encodeWord8 2, serialize' port, toCBOR (IP.fromIPv6 ip6)]
+    where
+      serialize' = toCBOR . toInteger
```

Comment:
It will be cleaner to avoid `mconcat`;
```hs
toCBOR (RelayAccessDomain domain port) =
     encodeListLen 3
  <> encodeWord8 0
  <> serialize' port
  <> toCBOR domain
toCBOR (RelayAccessAddress (IP.IPv4 ipv4) port) =
     encodeListLen 3
  <> encodeWord8 1
  <> serialize' port
  <> toCBOR (IP.fromIPv4 ipv4)
toCBOR (RelayAccessAddress (IP.IPv6 ip6) port) =
     encodeListLen 3
  <> encodeWord8 2
  <> serialize' port
  <> toCBOR (IP.fromIPv4 ip6)
```